### PR TITLE
Add zip archive distribution for libraries

### PR DESCRIPTION
**Pull request by ThadHouse**
Takes all artifacts and zips them into a zip with the correct headers and binaries.

Partial work on #611, still much more to go.

### CODE REVIEW

File: .azure/azure-pipelines.ci.yml [hunk 1]
```diff
@@ -359,3 +359,16 @@ stages:
...
+#
+# Distribution
+#
+
+- stage: distribution
+  displayName: Distribution
+  dependsOn:
+  - build_windows
+  - build_linux
+  - build_winkernel
```
Comment (nibanks) [inline .azure/azure-pipelines.ci.yml:372]: No need to worry about kernel mode stuff.

Comment (nibanks) [reply .azure/azure-pipelines.ci.yml:372]: Remove the dependency on build_winkernel too; and all the winkernel references in the distribution script.

**Comment by nibanks** [comment]
I feel like this should only run on master/release branches. Not PRs.

**Comment by ThadHouse** [comment]
Done, this now only runs on non prs.

**Pull request merged**